iceoryx2-bb/container/src/slotmap.rs

```diff
@@ -233,18 +233,15 @@ pub mod details {
         }
 
         pub(crate) unsafe fn insert_impl(&mut self, value: T) -> Option<SlotMapKey> {
-            match self.acquire_next_free_index() {
-                None => None,
-                Some(key) => {
-                    let key = SlotMapKey(key);
-                    self.store_value(key, value);
-                    Some(key)
-                }
-            }
+            self.acquire_next_free_index()
+                .map(|key| SlotMapKey(key))
+                .inspect(|key| {
+                    self.store_value(*key, value);
+                })
```

Comment:
Oh no, `inspect` was introduced with Rust 1.76 and our MSRV is 1.75.
```suggestion
            self.acquire_next_free_index().map(|key| {
                let key = SlotMapKey(key);
                self.store_value(key, value);
                key
            })
```